src/plugins/workspace/server/workspace_client.ts

```diff
@@ -122,6 +125,15 @@ export class WorkspaceClient implements IWorkspaceClientImpl {
           permissions,
         }
       );
+      if (dataSources && this.uiSettings && client) {
+        // This is for setting in workspace environment, otherwise uiSettings can't set workspace level value.
+        updateWorkspaceState(requestDetail.request, {
```

Comment:
I feel a little bit uncomfortable here as workspaceState should be set in the preRoute liftCycle. Can we as least reset the workspaceState once we finish `checkAndSetDefaultDataSources`?